exec/java-exec/src/main/codegen/templates/TypeHelper.java

```diff
@@ -68,8 +68,23 @@ public static JType getHolderType(JCodeModel model, MinorType type, DataMode mod
     case UNION:
       return model._ref(UnionHolder.class);
     case MAP:
+        switch (mode) {
+          case REQUIRED:
```

Comment:
Perhaps combine the case statements to avoid repeated code:

    case REQUIRED:
    case OPTIONAL:
        return model...